src/control/server/harness_test.go

```diff
@@ -260,20 +372,46 @@ func TestServer_HarnessIOServerStart(t *testing.T) {
 		},
 		"delayed failure": {
 			trc: &ioserver.TestRunnerConfig{
-				ErrChanCb: func() error {
-					time.Sleep(10 * time.Millisecond)
-					return errors.New("oops")
+				ErrChanCb: func(idx uint32) ioserver.InstanceError {
+					time.Sleep(testShortTimeout / 2)
+					return ioserver.InstanceError{Idx: idx, Err: errors.New("oops")}
 				},
 			},
-			expStartErr:   errors.New("oops"),
+			instanceUuids: map[int]string{
+				0: MockUUID(0),
+				1: MockUUID(1),
+			},
+			expStartErr:   context.DeadlineExceeded,
 			expStartCount: maxIOServers,
+			expDrpcCalls: map[uint32][]int32{
+				0: {
+					drpc.MethodSetRank,
+					drpc.MethodSetUp,
+				},
+				1: {
+					drpc.MethodSetRank,
+					drpc.MethodSetUp,
+				},
+			},
+			expGrpcCalls: map[uint32][]string{
+				0: {fmt.Sprintf("Join %d", system.NilRank)},
+				1: {fmt.Sprintf("Join %d", system.NilRank)},
+			},
+			expRanks: map[uint32]system.Rank{
+				0: system.Rank(0),
+				1: system.Rank(1),
+			},
+			expIoErrs: map[uint32]error{
+				0: errors.New("oops"),
+				1: errors.New("oo"),
```

Comment:
Should be "oops"?

Reply:
yes it should . but this actually passes because CmpErr only checks that the text is a substring of the error, not exact match.